AK/Math.h

```diff
@@ -8,249 +8,55 @@
 
 #include <AK/BuiltinWrappers.h>
 #include <AK/Concepts.h>
+#include <AK/NumericLimits.h>
 #include <AK/StdLibExtraDetails.h>
 #include <AK/Types.h>
+#include <AK/bits/Math/common.h>
 
 namespace AK {
 
-template<FloatingPoint T>
-constexpr T NaN = __builtin_nan("");
-template<FloatingPoint T>
-constexpr T Pi = 3.141592653589793238462643383279502884L;
-template<FloatingPoint T>
-constexpr T E = 2.718281828459045235360287471352662498L;
-
-namespace Details {
-template<size_t>
-constexpr size_t product_even();
-template<>
-constexpr size_t product_even<2>() { return 2; }
-template<size_t value>
-constexpr size_t product_even() { return value * product_even<value - 2>(); }
-
-template<size_t>
-constexpr size_t product_odd();
-template<>
-constexpr size_t product_odd<1>() { return 1; }
-template<size_t value>
-constexpr size_t product_odd() { return value * product_odd<value - 2>(); }
-}
-
-#define CONSTEXPR_STATE(function, args...)        \
-    if (is_constant_evaluated()) {                \
-        if (IsSame<T, long double>)               \
-            return __builtin_##function##l(args); \
-        if (IsSame<T, double>)                    \
-            return __builtin_##function(args);    \
-        if (IsSame<T, float>)                     \
-            return __builtin_##function##f(args); \
-    }
-
 namespace Division {
 template<FloatingPoint T>
-constexpr T fmod(T x, T y)
-{
-    CONSTEXPR_STATE(fmod, x, y);
-    u16 fpu_status;
-    do {
-        asm(
-            "fprem\n"
-            "fnstsw %%ax\n"
-            : "+t"(x), "=a"(fpu_status)
-            : "u"(y));
-    } while (fpu_status & 0x400);
-    return x;
-}
+CONSTEXPR T fmod(T x, T y);
```

Comment:
Declaring a function `constexpr` without giving it an actual definition doesn't accomplish anything.

If the definition comes from the architecture-specific headers, why bother declaring them here as well?

Or we can just forward to `cmath`'s `std` namespaced functions that already support overloading.

Reply:
To have a easy to find list of all functions supported/needed to be implemented

Reply:
> To have a easy to find list of all functions supported/needed to be implemented

Is that not what comments are for? The CONSTEXPR macro _is_ a bit confusing, mostly because the true purpose is not clear. So if the main reason is to flag code that needs to be refactored, then maybe it should be given a second round of thought?

Reply:
The Macro was removed....